contrib/cindercompatibleapi/api/attachment_test.go

```diff
@@ -0,0 +1,320 @@
+// Copyright (c) 2018 Huawei Technologies Co., Ltd. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+
+	"github.com/astaxie/beego"
+	c "github.com/opensds/opensds/client"
+	"github.com/opensds/opensds/contrib/cindercompatibleapi/converter"
+)
+
+func init() {
+	beego.Router("/V3/attachments/:attachmentId", &AttachmentPortal{},
+		"get:GetAttachment;delete:DeleteAttachment;put:UpdateAttachment")
+	beego.Router("/V3/attachments/detail", &AttachmentPortal{},
+		"get:ListAttachmentsDetails")
+
+	beego.Router("/V3/attachments", &AttachmentPortal{},
+		"post:CreateAttachment;get:ListAttachments")
+	if false == IsFakeClient {
+		client = NewFakeClient(&c.Config{Endpoint: TestEp})
+	}
+}
+
+////////////////////////////////////////////////////////////////////////////////
+//                            Tests for Attachment                              //
+////////////////////////////////////////////////////////////////////////////////
+func TestGetAttachment(t *testing.T) {
+	r, _ := http.NewRequest("GET", "/V3/attachments/f2dda3d2-bf79-11e7-8665-f750b088f63e", nil)
```

Comment:
Has this API been tested? Cinder (and I think all OpenStack) API's use lower case "v" in the version.